examples/pod_attach.rs

```diff
@@ -0,0 +1,122 @@
+                "command": ["sh", "-c", "for i in `seq 15`; do if [ $i -lt 7 ]; then echo \"o $i\"; else echo \"e $i\" 1>&2; fi; sleep 1; done;"],
+            }],
+        }
+    }))?;
+
+    let pods: Api<Pod> = Api::namespaced(client, &namespace);
+    // Stop on error including a pod already exists or is still being deleted.
+    pods.create(&PostParams::default(), &p).await?;
+
+    // Wait until the pod is running, otherwise we get 500 error.
+    let lp = ListParams::default().fields("metadata.name=example").timeout(10);
+    let mut stream = pods.watch(&lp, "0").await?.boxed();
+    while let Some(status) = stream.try_next().await? {
+        match status {
+            WatchEvent::Added(o) => {
+                info!("Added {}", Meta::name(&o));
+            }
+            WatchEvent::Modified(o) => {
+                let s = o.status.as_ref().expect("status exists on pod");
+                if s.phase.clone().unwrap_or_default() == "Running" {
+                    info!("Ready to attach to {}", Meta::name(&o));
+                    break;
+                }
+            }
+            _ => {}
+        }
+    }
+
+    let ap = AttachParams::default();
+    // Attach to see numbers printed on stdout.
+    let attached = pods.attach("example", &ap).await?;
+    // Separate stdout/stderr outputs
+    separate_outputs(attached).await;
+    // Combining stdout and stderr output.
+    // combined_output(proc).await;
+
+    // Delete it
+    pods.delete("example", &DeleteParams::default())
+        .await?
+        .map_left(|pdel| {
+            assert_eq!(Meta::name(&pdel), "example");
+        });
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+async fn separate_outputs(mut attached: AttachedProcess) {
+    let stdout_stream = attached.stdout().take().unwrap();
+    let stdouts = stdout_stream
+        .for_each(|out| {
+            if let Ok(bytes) = out {
+                let out = std::io::stdout();
+                out.lock().write_all(&bytes[..]).unwrap();
+            }
+            future::ready(())
+        })
+        .fuse();
+    let stderr_stream = attached.stderr().take().unwrap();
+    let stderrs = stderr_stream
+        .for_each(|out| {
+            if let Ok(bytes) = out {
+                let out = std::io::stderr();
+                out.lock().write_all(&bytes[..]).unwrap();
+            }
+            future::ready(())
+        })
+        .fuse();
+
+    join!(stdouts, stderrs);
+
+    if let Some(status) = attached.await {
+        println!("{:?}", status);
+    }
+}
+
+#[allow(dead_code)]
+async fn combined_output(mut attached: AttachedProcess) {
+    let stdout_stream = attached.stdout().take().unwrap();
+    let stderr_stream = attached.stderr().take().unwrap();
+    let outputs = select(stdout_stream, stderr_stream).for_each(|out| {
+        if let Ok(bytes) = out {
+            let out = std::io::stdout();
+            out.lock().write_all(&bytes[..]).unwrap();
+        }
+        future::ready(())
+    });
```

Comment:
the awkward `future::ready` can probably be avoided here if we use a for loop instead of a `for_each`, might be easier for the examples

Reply:
Fixed by using async blocks to keep it similar to the `separate_outputs` example above.

Reply:
Nice, that is a lot cleaner.